File: octopoints_flutter/lib/service/impl/UserServiceImpl.dart
import 'package:octopoints_flutter/db/DbSingleton.dart';
import 'package:octopoints_flutter/db/dao/UserDao.dart';
import 'package:octopoints_flutter/db/entity/UserEntity.dart';
import 'package:octopoints_flutter/service/UserService.dart';
import 'package:octopoints_flutter/service/model/User.dart';
import 'package:octopoints_flutter/service/mapper/UserMapper.dart';

class UserServiceImpl implements UserService {
  final UserDao _userDao = DbSingleton().db.userDao;

  @override
  Future<User> createUser(User user) async {
    int id = await _userDao.create(user.toUserEntity());
    return user.setId(id);
  }

  @override
  Future<int> deleteUser(User user) {
    return _userDao.remove(user.toUserEntity());
  }

  @override
  Future<User> getUserById(int id) async {
    UserEntity? userEntity = await _userDao.getUserById(id);
    return userEntity!.toUserModel();
  }

  @override
  Future<List<User>> getUsers() async {
    return (await _userDao.getAllUsers()).map((e) => e.toUserModel()).toList();
  }

  @override
  Future<int> updateUsers(List<User> users) {
    return _userDao.modify(users.map((e) => e.toUserEntity()).toList());
  }
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/impl/UserServiceImpl.kt
package dev.tpavani.octopoints.service.impl

import dev.tpavani.octopoints.db.DbSingleton
import dev.tpavani.octopoints.service.UserService
import dev.tpavani.octopoints.service.model.User
import dev.tpavani.octopoints.service.utils.ModelMapper

class UserServiceImpl: UserService {
    override suspend fun createUser(user: User): User {
        var id: Int = DbSingleton.getDb().userDao().create(ModelMapper.toUserEntity(user))
        return user.setId(id)
    }

    override suspend fun deleteUser(user: User) {
        DbSingleton.getDb().userDao().remove(ModelMapper.toUserEntity(user))
    }

    override suspend fun getUsers(): List<User> {
        return DbSingleton.getDb().userDao().getAllUsers().map { e -> ModelMapper.toUserModel(e)}
    }

    override suspend fun updateUser(users: List<User>) {
        DbSingleton.getDb().userDao().modify(users.map { e-> ModelMapper.toUserEntity(e) })
    }

}